.github/actions/artifact_upload/action.yml

```diff
@@ -0,0 +1,50 @@
+name: Upload artifact
+description: Upload an encrypted zip archive as a github artifact.
+inputs:
+  path:
+    description: 'The path(s) that should be uploaded.'
+    required: true
+  name:
+    description: 'The name of the artifact.'
+    required: true
+  retention-days:
+    description: 'How the artifact should be retained.'
+    required: false
+    default: 60
+  encryption-secret:
+    description: 'The secret to use for encrypting the files.'
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - name: Install zip
+      uses: montudor/action-zip@v1
+    - name: Find glob pattern matches
+      uses: tj-actions/glob@v17
+      id: glob
+      with:
+        files: |
+          ${{ inputs.path }}
```

Comment:
For this, I think we should be able to scrap it completely, normal bash globs should work.

Reply:
I don't think that normal bash globs would be enough since I cannot find a way to reliably apply multiple glob patterns while still allowing the same intuitive usage of them as we did before. For example:
https://github.com/edgelesssys/constellation/blob/e105ab5ced130d84ebb6be3df1d4ea1a61031d18/.github/actions/constellation_create/action.yml#L232-L234
Just appending these patterns to a command will not work, it would still zip the `terraform.log` file. I also wasn't able to find another way of applying both filters.

Reply:
`extglob` patterns are now used and it seems to work fine.